### ISSUE TITLE
Knight Household list - Issues and fixes

### ISSUE THREAD

**Issue by BSDataAnon** [BSData/adeptus-titanicus#195]
**File:** Household.catz

**BattleScribe version:** 2.03.15

**Platform:** iPhone / iPod / iPad

**Dropbox:** No

**Description:** So you can’t make a Household roster. You can only access Freeblades, and you cannot select a Baronical Court, leading to an unsolvable error.

**Comment by Prioress** [comment]
#199

**Comment by zopha** [comment]
May as well use this to start with. Currently the knight list it appears you can cannot select a Baronical court but there is an error saying you must add it but it cannot be selected.

![image](https://user-images.githubusercontent.com/1666415/150634820-b970ef07-4ef4-406c-8b0d-a19b4d0998fa.png)

The option to build a court is there but obscure so some changes will need to be done to get this working right. I think it will be fairly minor but I'm not very familar with knight households.

How to start a knight army (animated GIF)
![javaw_nhPHULQuIA](https://user-images.githubusercontent.com/1666415/150635013-65235be7-2770-4711-bdce-9e27f1a9a5a8.gif)

How it's structured in the data files
![image](https://user-images.githubusercontent.com/1666415/150635027-3dc7995a-2c57-4aaa-bec1-a6a5eedcd12e.png)

**Comment by zopha** [comment]
Tried duplicating the Lance entry and got a ton errors. Will look at this later or if someone else has time please go ahead and see what can be done.

![image](https://user-images.githubusercontent.com/1666415/150635101-75b78b0e-29ae-4e87-b563-3d99383259f1.png)

**Comment by Prioress** [comment]
Sorry for not answering for a while, I’ve been very busy learning how to
code 0-0

**Comment by MoDLegion** [comment]
You actually need to add a Lance force in the roster selection bit in order to create an actual household list. By default if you only select the 'household' force you can only add the freeblades and support stuff. Once you have an actual Lance in your force you can add the baronial court and other banners to it etc.

I did noticed there are still 'structure' selections when adding banners to a lance and then it starts complaining that you dont have a structure selection, but when you do make a structure selection it then complains that you are not allowed to. All in all the household list part seems to be working, in part, but can definitely use some re-engineering to make it more user friendly.

**Comment by MoDLegion** [comment]
I think all that really needs to be done is the following:
- [ ] Remove structure point tracking from Lance banners (it's already gone from support banners)
- [ ] Add Knight Armiger Banner (Helverins and Warglaives)
- [ ] Add Knights Moirax Banner

**Comment by zopha** [comment]
Review Knight Asterius would be another. The model included some slightly different rules when FW released it. I have them somewhere so can do it or list the changes for it.

**Comment by MoDLegion** [comment]
Hmm yeah, it also does not seem to allow any Knight Acastus selections, in neither a Battlegroup nor a Household list. There is modifier on it for each selection of Lance in Force. If you can list the changes I can put them in as I've half begun fixing issues in my investigation to find out what else is broken (I think the only change is the twin volkite culverins being the only hull weapon option and that Acastus' can now be part of the Baronial Court).

I don't have the rules for Households, but according to Goonhammer 'Each Lance consists of three Banners, two of which must be the same chassis' this requirement also isn't implemented. I think this is what the modifier on each Lance was for, but they all don't reference anything. So far I've discovered the following extra issues:

- [ ] Reivew Knight Acastus Asterius (hull weapon/Baronial court)
- [ ] Fix Knight Acastus Banner maximum selection validation
- [ ] Implement Lance Banner chassis restriction
- [ ] Warhound Titan support selection is broken

**Comment by MoDLegion** [comment]
I ended up copying some of the Armiger profile work done by #219, it wasn't really possible to integrate the work in that PR directly, but it also was missing a lot of things. Implementation of the Knight Asterius are already tracked in a separate issue so I opted to leave that out of my PR. The only issue I wasn't able to resolve is getting the validation for needing a Lance in a Household, but instead I updated the readme with clear instructions on how to properly create either a Legio Battlegroup or a Household Battlegroup which should hopefully resolve the initial complaint that lead to this issue.
